windows/utils/get_windows_system_info.yml

```diff
@@ -35,8 +36,16 @@
 - name: "Get Windows guest OS edition from distribution info"
   ansible.builtin.set_fact:
     guest_os_edition: >-
-      {{ (guest_os_ansible_distribution | regex_search('Microsoft Windows( Server)? ([0-9]+) (R2 )?(.*)', '\4') | first | regex_replace('insider|preview', '', ignorecase=True)).strip() }}
-  when: guest_os_ansible_distribution
+      {{ (guest_os_ansible_distribution | regex_search('Windows( Server)? ([0-9]+) (R2 )?(.*)', '\4') | first | regex_replace('insider|preview', '', ignorecase=True)).strip() }}
+  when:
+    - guest_os_ansible_distribution | regex_search('Windows( Server)? ([0-9]+) (R2 )?(.*)')
```

Comment:
Here can use `vars` to reduce duplicate expression like:
```
    - name: "Get Windows guest OS edition from distribution info"
      ansible.builtin.set_fact:
        guest_os_edition: >-
          {{ (_windows_edition | first | regex_replace('insider|preview', '', ignorecase=True)).strip() }}
      when: _windows_edition
      vars:
        _windows_edition: "{{ guest_os_ansible_distribution | regex_search('Windows( Server)? ([0-9]+) (R2 )?(.*)', '\\4') }}"
```

And same to below `guest_os_product_version` fact setting.